x/ecocredit/params.go

```diff
@@ -36,6 +36,27 @@ func (p *Params) ParamSetPairs() paramtypes.ParamSetPairs {
 	}
 }
 
+// Validate will run each param field's validate method
+func (p Params) Validate() error {
+	if err := validateCreditTypes(p.CreditTypes); err != nil {
+		return err
+	}
+
+	if err := validateAllowedClassCreators(p.AllowedClassCreators); err != nil {
```

Comment:
Although those functions are already tested independently, it'd be good still to add tests for the err cases within `Params.Validate` IMO

Reply:
ok, added some more cases here https://github.com/regen-network/regen-ledger/pull/511/commits/0201ea829e05c39a1e811bb939bd415ddf9848fa